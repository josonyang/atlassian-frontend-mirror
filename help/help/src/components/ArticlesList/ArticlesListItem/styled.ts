/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import styled from '@emotion/styled';
import * as colors from '@atlaskit/theme/colors';
import { fontFallback } from '@atlaskit/theme/typography';
import { token } from '@atlaskit/tokens';

type ArticlesListItemWrapperProps = {
	styles: any;
};

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
export const ArticlesListItemWrapper = styled.a<ArticlesListItemWrapperProps>(
	{
		position: 'relative',
		boxSizing: 'border-box',
		padding: token('space.100', '8px'),
		display: 'block',
		textDecoration: 'none',
		cursor: 'pointer',
		// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
		color: token('color.text.subtlest', colors.N200),
		borderRadius: '3px',
		backgroundColor: token('color.background.neutral.subtle', 'transparent'),

		'&:hover, &:focus, &:visited, &:active': {
			textDecoration: 'none',
			outline: 'none',
			outlineOffset: 'none',
		},

		'&:focus': {
			// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
			boxShadow: `${token('color.border.focused', colors.B100)} 0px 0px 0px 2px inset`,
		},

		'&:hover': {
			// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
			backgroundColor: token('color.background.neutral.subtle.hovered', colors.N30),
		},

		'&:active': {
			// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
			backgroundColor: token('color.background.neutral.subtle.pressed', colors.B50),
		},
	},
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-dynamic-styles -- Ignored via go/DSP-18766
	(props: any) => props.styles,
);

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
export const ArticlesListItemContainer = styled.div({
	width: '100%',
	whiteSpace: 'nowrap',
	display: 'flex',
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles, @atlaskit/ui-styling-standard/no-styled -- Ignored via go/DSP-18766
export const ArticlesListItemTitleSection = styled.div({
	display: 'flex',
	flexDirection: 'column',
	flexGrow: 1,
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
export const ArticlesListItemTypeTitle = styled.div({
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
	font: token('font.body.small', fontFallback.body.small),
	fontWeight: token('font.weight.bold', 'bold'),
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
	color: token('color.text.subtlest', colors.N200),
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
export const ArticlesListItemLinkIcon = styled.span({
	alignSelf: 'auto',
	paddingInlineStart: token('space.050', '4px'),
	verticalAlign: 'middle',
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
export const ArticlesListItemTitleText = styled.p({
	textDecoration: 'none',
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
	color: token('color.text', colors.N800),
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
	font: token('font.heading.xsmall', fontFallback.heading.xsmall),
	display: 'inline-block',
	whiteSpace: 'normal',
	overflow: 'hidden',
	marginBottom: token('space.100', '4px'),
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
export const ArticlesListItemDescription = styled.p({
	display: 'block',
	lineHeight: '20px',
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
	color: token('color.text.subtle', colors.N400),
	margin: 0,
	paddingBottom: token('space.025', '2px'),
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles, @atlaskit/ui-styling-standard/no-styled -- Ignored via go/DSP-18766
export const ArticlesListItemSource = styled.div({
	display: 'flex',
	alignItems: 'center',
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
	font: token('font.heading.xxsmall', fontFallback.heading.xxsmall),
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
	color: token('color.text.subtlest', colors.N400A),
	padding: `${token('space.050', '4px')} 0`,
	fontWeight: token('font.weight.bold', 'bold'),
	textTransform: 'uppercase',
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles, @atlaskit/ui-styling-standard/no-styled -- Ignored via go/DSP-18766
export const ArticlesListItemTrustFactor = styled.div({
	display: 'flex',
	alignItems: 'center',
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
	font: token('font.body.small', fontFallback.body.small),
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
	color: token('color.text.subtlest', colors.N400A),
	paddingTop: token('space.025', '2px'),
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles, @atlaskit/ui-styling-standard/no-styled -- Ignored via go/DSP-18766
export const ArticlesListItemViewCount = styled.span({
	paddingRight: token('space.100', '8px'),
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles, @atlaskit/ui-styling-standard/no-styled -- Ignored via go/DSP-18766
export const ArticlesListItemHelpfulCount = styled.span({
	display: 'inline-flex',
	paddingRight: token('space.100', '8px'),
	alignItems: 'center',
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors
	'& > span': {
		marginRight: token('space.025', '2px'),
	},
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles, @atlaskit/ui-styling-standard/no-styled -- Ignored via go/DSP-18766
export const ArticlesListItemLastModified = styled.div({
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
	font: token('font.body.small', fontFallback.body.small),
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
	color: token('color.text.subtlest', colors.N400A),
	padding: `${token('space.050', '4px')} 0`,
});
