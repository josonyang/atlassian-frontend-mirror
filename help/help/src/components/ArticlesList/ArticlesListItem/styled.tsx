/**
 * @jsxRuntime classic
 * @jsx jsx
 */

import React from 'react';
import { css, jsx } from '@compiled/react';
import Heading from '@atlaskit/heading';
import { fontFallback } from '@atlaskit/theme/typography';
import { token } from '@atlaskit/tokens';
import { B100, B50, N200, N30, N400, N400A, N800, Y100 } from '@atlaskit/theme/colors';

const { heading, body } = fontFallback;

const articlesListItemWrapperStyles = css({
	position: 'relative',
	boxSizing: 'border-box',
	paddingTop: token('space.100', '8px'),
	paddingRight: token('space.100', '8px'),
	paddingBottom: token('space.100', '8px'),
	paddingLeft: token('space.100', '8px'),
	display: 'block',
	textDecoration: 'none',
	cursor: 'pointer',
	color: token('color.text.subtlest', N200),
	// eslint-disable-next-line @atlaskit/design-system/no-unsafe-design-token-usage
	borderRadius: token('border.radius.100', '3px'),
	backgroundColor: token('color.background.neutral.subtle', 'transparent'),
	'&:hover, &:focus, &:visited, &:active': {
		textDecoration: 'none',
		outline: 'none',
		outlineOffset: 'none',
	},
	'&:focus': {
		boxShadow: `${token('color.border.focused', B100)} 0px 0px 0px 2px inset`,
	},
	'&:hover': {
		backgroundColor: token('color.background.neutral.subtle.hovered', N30),
	},
	'&:active': {
		backgroundColor: token('color.background.neutral.subtle.pressed', B50),
	},
});

export const ArticlesListItemWrapper = ({
	styles,
	children,
	...rest
}: {
	children: React.ReactNode;
	styles?: React.CSSProperties;
	[rest: string]: any;
}) => (
	// eslint-disable-next-line @atlaskit/design-system/no-html-anchor, @atlaskit/ui-styling-standard/enforce-style-prop
	<a css={articlesListItemWrapperStyles} style={styles} {...rest}>
		{children}
	</a>
);

const articlesListItemContainerStyles = css({
	width: '100%',
	whiteSpace: 'nowrap',
	display: 'flex',
});

export const ArticlesListItemContainer = ({ children }: { children: React.ReactNode }) => (
	// eslint-disable-next-line @atlaskit/design-system/use-primitives
	<div css={articlesListItemContainerStyles}>{children}</div>
);

const articlesListItemTitleSectionStyles = css({
	display: 'flex',
	flexDirection: 'column',
	flexGrow: 1,
});

export const ArticlesListItemTitleSection = ({ children }: { children: React.ReactNode }) => (
	<div css={articlesListItemTitleSectionStyles}>{children}</div>
);

const articlesListItemLinkIconStyles = css({
	alignSelf: 'auto',
	paddingInlineStart: token('space.050', '4px'),
	verticalAlign: 'middle',
});

export const ArticlesListItemLinkIcon = ({ children }: { children: React.ReactNode }) => (
	<span css={articlesListItemLinkIconStyles}>{children}</span>
);

const articlesListItemTitleTextStyles = css({
	textDecoration: 'none',
	color: token('color.text', N800),
	font: token('font.heading.xsmall', heading.xsmall),
	display: 'inline-block',
	whiteSpace: 'normal',
	overflow: 'hidden',
	marginBottom: token('space.100', '4px'),
});

export const ArticlesListItemTitleText = ({ children }: { children: React.ReactNode }) => (
	<div css={articlesListItemTitleTextStyles}>
		<Heading size="xsmall" as="h3">
			{children}
		</Heading>
	</div>
);

const articlesListItemDescriptionStyles = css({
	display: 'block',
	// eslint-disable-next-line @atlaskit/design-system/use-tokens-typography
	lineHeight: '20px',
	color: token('color.text.subtle', N400),
	margin: 0,
	paddingBottom: token('space.025', '2px'),
});

export const ArticlesListItemDescription = ({ children }: { children: React.ReactNode }) => (
	// eslint-disable-next-line @atlaskit/design-system/use-primitives-text
	<p css={articlesListItemDescriptionStyles}>{children}</p>
);

const articlesListItemSourceStyles = css({
	display: 'flex',
	alignItems: 'center',
	font: token('font.heading.xxsmall', heading.xxsmall),
	color: token('color.text.subtlest', N400A),
	paddingTop: token('space.050', '4px'),
	paddingRight: 0,
	paddingBottom: token('space.050', '4px'),
	paddingLeft: 0,
	fontWeight: token('font.weight.bold', 'bold'),
	// eslint-disable-next-line @atlaskit/design-system/use-tokens-typography
	textTransform: 'uppercase',
});

export const ArticlesListItemSource = ({ children }: { children: React.ReactNode }) => (
	// eslint-disable-next-line @atlaskit/design-system/use-primitives-text
	<div css={articlesListItemSourceStyles}>{children}</div>
);

const articlesListItemTrustFactorStyles = css({
	display: 'flex',
	alignItems: 'center',
	font: token('font.body.small', body.small),
	color: token('color.text.subtlest', N400A),
	paddingTop: token('space.025', '2px'),
});

export const ArticlesListItemTrustFactor = ({ children }: { children: React.ReactNode }) => (
	<div css={articlesListItemTrustFactorStyles}>{children}</div>
);

const articlesListItemViewCountStyles = css({
	paddingRight: token('space.100', '8px'),
});

export const ArticlesListItemViewCount = ({ children }: { children: React.ReactNode }) => (
	<span css={articlesListItemViewCountStyles}>{children}</span>
);

const articlesListItemHelpfulCountStyles = css({
	display: 'inline-flex',
	paddingRight: token('space.100', '8px'),
	alignItems: 'center',
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors
	'& > span': {
		marginRight: token('space.025', '2px'),
	},
});

export const ArticlesListItemHelpfulCount = ({ children }: { children: React.ReactNode }) => (
	<span css={articlesListItemHelpfulCountStyles}>{children}</span>
);

const articlesListItemLastModifiedStyles = css({
	font: token('font.body.small', body.small),
	color: token('color.text.subtlest', N400A),
	paddingTop: token('space.050', '4px'),
	paddingRight: 0,
	paddingBottom: token('space.050', '4px'),
	paddingLeft: 0,
});

export const ArticlesListItemLastModified = ({ children }: { children: React.ReactNode }) => (
	<div css={articlesListItemLastModifiedStyles}>{children}</div>
);

const articlesListItemDescriptionHighlightStyles = css({
	backgroundColor: token('color.background.accent.yellow.subtlest', Y100),
});

export const ArticlesListItemDescriptionHighlight = ({
	key,
	children,
}: {
	key: number | string;
	children: React.ReactNode;
}) => (
	<mark key={key} css={articlesListItemDescriptionHighlightStyles}>
		{children}
	</mark>
);
