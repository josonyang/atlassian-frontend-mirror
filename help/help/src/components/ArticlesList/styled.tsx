/**
 * @jsxRuntime classic
 * @jsx jsx
 */

import React from 'react';
import { css, jsx } from '@compiled/react';
import { token } from '@atlaskit/tokens';

const articlesListContainerStyles = css({
	position: 'relative',
});

export const ArticlesListContainer = ({ children }: { children: React.ReactNode }) => (
	<div css={articlesListContainerStyles}>{children}</div>
);

const toggleShowMoreArticlesContainerStyles = css({
	paddingTop: token('space.100', '8px'),
	paddingRight: 0,
	paddingBottom: token('space.100', '8px'),
	paddingLeft: 0,
	// eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
	span: {
		margin: 0,
	},
});

export const ToggleShowMoreArticlesContainer = ({ children }: { children: React.ReactNode }) => (
	<div css={toggleShowMoreArticlesContainerStyles}>{children}</div>
);
