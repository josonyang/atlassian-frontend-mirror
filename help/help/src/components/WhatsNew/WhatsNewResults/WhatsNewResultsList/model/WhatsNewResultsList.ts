import { type WhatsNewArticleItem } from '../../../../../model/WhatsNew';
import type UIAnalyticsEvent from '@atlaskit/analytics-next/UIAnalyticsEvent';

export interface WhatsNewResultsList {
	hasNextPage?: boolean;
	loadingMore?: boolean;
	nextPage?: string;
	onShowMoreButtonClick?: (
		event: React.MouseEvent<HTMLElement>,
		analyticsEvent: UIAnalyticsEvent,
	) => void;
	onWhatsNewResultItemClick?: (
		event: React.MouseEvent<HTMLElement, MouseEvent>,
		analyticsEvent: UIAnalyticsEvent,
		articleData: WhatsNewArticleItem,
	) => void;
	// Style. This component has two different styles (primary and secondary)
	style?: 'primary' | 'secondary';
	/* List of "What's New" items. This prop is optional */
	whatsNewArticles?: WhatsNewArticleItem[] | null;
}
