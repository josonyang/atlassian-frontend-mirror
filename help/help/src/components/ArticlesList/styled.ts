/** @jsx jsx */

import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { token } from '@atlaskit/tokens';

export const truncate = (width: string = '100%') =>
  css({
    overflowX: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',
    width: width,
  });

export const ArticlesListContainer = styled.div({
  position: 'relative',
});

export const ToggleShowMoreArticlesContainer = styled.div({
  padding: `${token('space.100', '8px')} 0`,
  span: {
    margin: 0,
  },
});
