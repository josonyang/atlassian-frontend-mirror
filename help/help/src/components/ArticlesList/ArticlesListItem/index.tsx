import React from 'react';
import * as colors from '@atlaskit/theme/colors';
import {
	useAnalyticsEvents,
	type UIAnalyticsEvent,
	AnalyticsContext,
} from '@atlaskit/analytics-next';
import ShortcutIcon from '@atlaskit/icon/glyph/shortcut';
import { token } from '@atlaskit/tokens';
import { injectIntl, type WrappedComponentProps } from 'react-intl-next';

import { type ArticleItem } from '../../../model/Article';
import LikeIcon from '@atlaskit/icon/glyph/like';
import {
	ArticlesListItemWrapper,
	ArticlesListItemContainer,
	ArticlesListItemTitleText,
	ArticlesListItemDescription,
	ArticlesListItemLinkIcon,
	ArticlesListItemTrustFactor,
	ArticlesListItemViewCount,
	ArticlesListItemHelpfulCount,
	ArticlesListItemSource,
	ArticlesListItemLastModified,
	ArticlesListItemTitleSection,
} from './styled';

const ANALYTICS_CONTEXT_DATA = {
	componentName: 'ArticlesListItem',
	packageName: process.env._PACKAGE_NAME_,
	packageVersion: process.env._PACKAGE_VERSION_,
};

interface Props {
	styles?: {};
	/* Function executed when the user clicks the related article */
	onClick?: (event: React.MouseEvent<HTMLElement>, analyticsEvent: UIAnalyticsEvent) => void;
}

export const ArticlesListItem: React.FC<Props & Partial<ArticleItem> & WrappedComponentProps> = ({
	styles,
	title,
	description,
	href = '',
	onClick,
	trustFactors,
	source,
	lastPublished,
}) => {
	const { createAnalyticsEvent } = useAnalyticsEvents();

	const handleOnClick = (event: React.MouseEvent<HTMLElement>): void => {
		event.preventDefault();
		if (onClick) {
			const analyticsEvent: UIAnalyticsEvent = createAnalyticsEvent({
				action: 'clicked',
			});

			onClick(event, analyticsEvent);
		}
	};

	// Check if trust factors are available
	const isNumViewsVisible = trustFactors?.numViews != null;
	const isHelpfulCountVisible = trustFactors?.helpfulCount != null;
	const isTrustFactorVisible = isNumViewsVisible || isHelpfulCountVisible;
	// Check if source is available
	const isSourceVisible = source != null;

	const isLastPublishedVisible = lastPublished != null && lastPublished !== '';

	return (
		<ArticlesListItemWrapper
			styles={styles}
			aria-disabled="false"
			role="button"
			href={href}
			onClick={handleOnClick}
		>
			<ArticlesListItemContainer>
				<ArticlesListItemTitleSection>
					<ArticlesListItemTitleText>{title}</ArticlesListItemTitleText>
					{isLastPublishedVisible && (
						<ArticlesListItemLastModified>
							Last modified: {lastPublished}
						</ArticlesListItemLastModified>
					)}
				</ArticlesListItemTitleSection>
				{href && (
					<ArticlesListItemLinkIcon>
						<ShortcutIcon
							size="small"
							label=""
							primaryColor={token('color.icon.subtle', colors.N90)}
							secondaryColor={token('color.icon.subtle', colors.N90)}
						/>
					</ArticlesListItemLinkIcon>
				)}
			</ArticlesListItemContainer>

			<ArticlesListItemDescription>{description}</ArticlesListItemDescription>
			{isSourceVisible && <ArticlesListItemSource>{source}</ArticlesListItemSource>}
			{isTrustFactorVisible && (
				<ArticlesListItemTrustFactor>
					{isNumViewsVisible && (
						<ArticlesListItemViewCount>{trustFactors.numViews} views</ArticlesListItemViewCount>
					)}
					{isHelpfulCountVisible && (
						<ArticlesListItemHelpfulCount>
							<LikeIcon label="Like" size="small" />
							{trustFactors.helpfulCount}
						</ArticlesListItemHelpfulCount>
					)}
				</ArticlesListItemTrustFactor>
			)}
		</ArticlesListItemWrapper>
	);
};

const ArticlesListItemWithContext: React.FC<
	Props & Partial<ArticleItem> & WrappedComponentProps
> = (props) => {
	return (
		<AnalyticsContext data={ANALYTICS_CONTEXT_DATA}>
			<ArticlesListItem {...props} />
		</AnalyticsContext>
	);
};

export default injectIntl(ArticlesListItemWithContext);
