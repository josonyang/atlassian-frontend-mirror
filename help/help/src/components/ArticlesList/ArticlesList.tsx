import React from 'react';
import { type UIAnalyticsEvent } from '@atlaskit/analytics-next';
import { N30 } from '@atlaskit/theme/colors';
import { token } from '@atlaskit/tokens';
import AnimateHeight from 'react-animate-height';

import { type ArticleItem } from '../../model/Article';

import ArticlesListItem from './ArticlesListItem';
import { type ArticlesList as ArticlesListInterface } from './model/ArticlesListItem';
import {
  MIN_ITEMS_TO_DISPLAY,
  ANIMATE_HEIGHT_TRANSITION_DURATION_MS,
} from './constants';

export interface Props {
  /* Number of articles to display. This prop is optional (default value is 5) */
  numberOfArticlesToDisplay?: number;
}

const getStyles = (style: string) => {
  if (style === 'secondary') {
    return {
      border: `2px solid ${token('color.border', N30)}`,
      padding: token('space.200', '16px'),
      marginBottom: token('space.150', '12px'),
    };
  }
  return {
    border: 0,
    padding: token('space.100', '8px'),
    marginBottom: 0,
  };
};

const articlesList: React.FC<Partial<ArticlesListInterface> & Props> = ({
  style = 'primary',
  articles = [],
  minItemsToDisplay = MIN_ITEMS_TO_DISPLAY,
  numberOfArticlesToDisplay = MIN_ITEMS_TO_DISPLAY,
  onArticlesListItemClick,
}) =>
  articles && (
    <>
      <div>
        {articles.slice(0, minItemsToDisplay).map((article: ArticleItem) => {
          return (
            <ArticlesListItem
              styles={getStyles(style)}
              id={article.id}
              onClick={(
                event: React.MouseEvent<HTMLElement>,
                analyticsEvent: UIAnalyticsEvent,
              ) => {
                if (onArticlesListItemClick) {
                  onArticlesListItemClick(event, analyticsEvent, article);
                }
              }}
              title={article.title}
              description={article.description}
              key={article.id}
              href={article.href}
              trustFactors={article.trustFactors}
              source={article.source}
              lastPublished={article.lastPublished}
            />
          );
        })}
      </div>
      <AnimateHeight
        duration={ANIMATE_HEIGHT_TRANSITION_DURATION_MS}
        easing="linear"
        height={numberOfArticlesToDisplay > minItemsToDisplay ? 'auto' : 0}
      >
        {articles
          .slice(minItemsToDisplay, articles.length)
          .map((article: ArticleItem) => (
            <ArticlesListItem
              styles={getStyles(style)}
              id={article.id}
              onClick={(
                event: React.MouseEvent<HTMLElement>,
                analyticsEvent: UIAnalyticsEvent,
              ) => {
                if (onArticlesListItemClick) {
                  onArticlesListItemClick(event, analyticsEvent, article);
                }
              }}
              title={article.title}
              description={article.description}
              key={article.id}
              href={article.href}
              trustFactors={article.trustFactors}
              source={article.source}
              lastPublished={article.lastPublished}
            />
          ))}
      </AnimateHeight>
    </>
  );

export default articlesList;
