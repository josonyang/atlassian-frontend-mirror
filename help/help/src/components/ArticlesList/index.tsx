import React, { useState } from 'react';
import { type UIAnalyticsEvent } from '@atlaskit/analytics-next';

import ArticlesList from './ArticlesList';
import ShowMoreButton from '../ShowMoreButton';
import { ArticlesListContainer } from './styled';
import { type ArticlesList as ArticlesListInterface } from './model/ArticlesListItem';
import { MIN_ITEMS_TO_DISPLAY } from './constants';

const ArticleList: React.FC<ArticlesListInterface> = ({
  style,
  articles,
  minItemsToDisplay,
  maxItemsToDisplay,
  onArticlesListItemClick,
  onToggleArticlesList,
}) => {
  const [showMoreToggled, setShowMoreToggled] = useState<boolean>(true);

  const getMinItemsToDisplay = (): number => {
    return minItemsToDisplay ? minItemsToDisplay : MIN_ITEMS_TO_DISPLAY;
  };

  const getMaxItemsToDisplay = (): number => {
    if (articles) {
      return maxItemsToDisplay ? maxItemsToDisplay : articles.length;
    }

    return 0;
  };

  const getNumberOfArticlesToDisplay = (showMoreToggeled: boolean): number => {
    return showMoreToggeled ? getMinItemsToDisplay() : getMaxItemsToDisplay();
  };

  const toggleArticlesList = (
    event: React.MouseEvent<HTMLElement>,
    analyticsEvent: UIAnalyticsEvent,
  ): void => {
    setShowMoreToggled(!showMoreToggled);

    if (onToggleArticlesList) {
      onToggleArticlesList(event, analyticsEvent, !showMoreToggled);
    }
  };

  return articles && articles.length > 0 ? (
    <ArticlesListContainer>
      <>
        <ArticlesList
// eslint-disable-next-line @atlaskit/ui-styling-standard/enforce-style-prop -- Ignored via go/DSP-18766
          style={style}
          onArticlesListItemClick={onArticlesListItemClick}
          articles={articles}
          numberOfArticlesToDisplay={getNumberOfArticlesToDisplay(
            showMoreToggled,
          )}
        />
        {articles.length > getMinItemsToDisplay() && (
          <ShowMoreButton
            minItemsToDisplay={getMinItemsToDisplay()}
            maxItemsToDisplay={getMaxItemsToDisplay()}
            showMoreToggeled={showMoreToggled}
            onToggle={toggleArticlesList}
            itemsType="articles"
          />
        )}
      </>
    </ArticlesListContainer>
  ) : null;
};

export default ArticleList;
