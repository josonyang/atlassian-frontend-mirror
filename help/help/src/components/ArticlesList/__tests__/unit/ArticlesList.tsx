import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl-next';

import { getMockArticleItemList } from '../../../../util/testing/mock';

import ArticlesList from '../../ArticlesList';
import { type ArticlesList as ArticlesListInterface } from '../../model/ArticlesListItem';

const mockOnArticlesListItemClick = jest.fn();
const mockOnToggleArticlesList = jest.fn();
let ArticlesListProps: Partial<ArticlesListInterface>;

describe('ArticleContent', () => {
	beforeEach(() => {
		ArticlesListProps = {
			articles: getMockArticleItemList(10),
			onArticlesListItemClick: mockOnArticlesListItemClick,
			onToggleArticlesList: mockOnToggleArticlesList,
		};
	});

	it('should capture and report a11y violations', async () => {
		const { container } = render(
			<IntlProvider locale="en">
				<ArticlesList {...ArticlesListProps} minItemsToDisplay={3} numberOfArticlesToDisplay={6} />
			</IntlProvider>,
		);

		await expect(container).toBeAccessible();
	});

	it.skip('Should match snapshot', () => {
		const { container } = render(
			<IntlProvider locale="en">
				<ArticlesList {...ArticlesListProps} />
			</IntlProvider>,
		);

		expect(container.firstChild).toMatchSnapshot();
	});
});
