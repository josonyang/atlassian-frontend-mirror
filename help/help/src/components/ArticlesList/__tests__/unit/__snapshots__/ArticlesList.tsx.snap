// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleContent Should match snapshot 1`] = `
.emotion-5 {
  position: relative;
  box-sizing: border-box;
  padding: var(--ds-space-100, 8px);
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: var(--ds-text-subtlest, #6B778C);
  border-radius: 3px;
  background-color: var(--ds-background-neutral-subtle, transparent);
  border: 0;
  padding: var(--ds-space-150, 12px);
  margin-bottom: 0;
}

.emotion-5:hover,
.emotion-5:focus,
.emotion-5:visited,
.emotion-5:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  outline-offset: none;
}

.emotion-5:focus {
  box-shadow: var(--ds-border-focused, #4C9AFF) 0px 0px 0px 2px inset;
}

.emotion-5:hover {
  background-color: var(--ds-background-neutral-subtle-hovered, #EBECF0);
}

.emotion-5:active {
  background-color: var(--ds-background-neutral-subtle-pressed, #DEEBFF);
}

.emotion-3 {
  width: 100%;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--ds-text, #172B4D);
  font: var(--ds-font-heading-xsmall, normal 600 14px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Ubuntu, system-ui, "Helvetica Neue", sans-serif);
  display: inline-block;
  white-space: normal;
  overflow: hidden;
  margin-bottom: var(--ds-space-100, 4px);
}

.emotion-1 {
  font: var(--ds-font-body-small, normal 400 11px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Ubuntu, system-ui, "Helvetica Neue", sans-serif);
  color: var(--ds-text-subtlest, rgba(9, 30, 66, 0.71));
  padding: var(--ds-space-050, 4px) 0;
}

.emotion-4 {
  display: block;
  line-height: 20px;
  color: var(--ds-text-subtle, #505F79);
  margin: 0;
  padding-bottom: var(--ds-space-025, 2px);
}

<div>
  <a
    aria-disabled="false"
    class="emotion-5"
    href=""
    role="button"
  >
    <div
      class="emotion-3"
    >
      <div
        class="emotion-2"
      >
        <p
          class="emotion-0"
        >
          mockArticle title
        </p>
        <div
          class="emotion-1"
        >
          Last modified: 
          2020-04-29T06:05:02.426Z
        </div>
      </div>
      
    </div>
    <p
      class="emotion-4"
    >
      mockArticle description
    </p>
  </a>
  <a
    aria-disabled="false"
    class="emotion-5"
    href=""
    role="button"
  >
    <div
      class="emotion-3"
    >
      <div
        class="emotion-2"
      >
        <p
          class="emotion-0"
        >
          mockArticle title
        </p>
        <div
          class="emotion-1"
        >
          Last modified: 
          2020-04-29T06:05:02.426Z
        </div>
      </div>
      
    </div>
    <p
      class="emotion-4"
    >
      mockArticle description
    </p>
  </a>
  <a
    aria-disabled="false"
    class="emotion-5"
    href=""
    role="button"
  >
    <div
      class="emotion-3"
    >
      <div
        class="emotion-2"
      >
        <p
          class="emotion-0"
        >
          mockArticle title
        </p>
        <div
          class="emotion-1"
        >
          Last modified: 
          2020-04-29T06:05:02.426Z
        </div>
      </div>
      
    </div>
    <p
      class="emotion-4"
    >
      mockArticle description
    </p>
  </a>
</div>
`;
